modules/local/starfusion/detect/main.nf

```diff
@@ -18,12 +18,13 @@ process STARFUSION {
     script:
     def prefix = task.ext.prefix ?: "${meta.id}"
     def fasta = meta.single_end ? "--left_fq ${reads[0]}" : "--left_fq ${reads[0]} --right_fq ${reads[1]}"
+    def junction_arg = junction ? "-J ${junction}": ""
     def args = task.ext.args ?: ''
     """
     STAR-Fusion \\
         --genome_lib_dir $reference \\
         $fasta \\
-        -J $junction \\
+        $junction_arg \\
```

Comment:
Now the module behaves as follows: if `fasta` is provided, it will use that, otherwise, if will be empty and the junction file will be used if provided.